src/libaktualizr/package_manager/packagemanagerinterface.cc

```diff
@@ -98,6 +110,8 @@ bool PackageManagerInterface::fetchTarget(const Uptane::Target& target, Uptane::
       return true;
     }
     std::unique_ptr<DownloadMetaStruct> ds = std_::make_unique<DownloadMetaStruct>(target, progress_cb, token);
+    ds->time_lastreport =
+        std::chrono::steady_clock::now();  // use this time to report download progress it download take long time
```

Comment:
Two minor issues:
1. I think this initialization should either be in the constructor of the `DownloadMetaStruct` or right before the download is initiated (before the loop with `http_->download` below).
2. If a comment is so long as to cause the code to require a linebreak, I think it would make sense to put it on the preceding line instead.

Otherwise looks good!

Reply:
done